core/src/App.html

```diff
@@ -304,181 +314,194 @@
           }
         });
         window.addEventListener('message', async e => {
-          if ('luigi.get-context' === e.data.msg) {
-            if (
-              this.get().modalIframe &&
-              e.source === this.get().modalIframe.contentWindow
-            ) {
-              let ctx = this.get().modalIframeData.context;
-              const modalConfig = {
-                ...config,
-                iframe: this.get().modalIframe,
-                context: ctx,
-                pathParams: this.get().modalIframeData.pathParams,
-                nodeParams: this.get().modalIframeData.nodeParams,
-                modal: true
-              };
-              sendContextToClient(this, modalConfig, {});
-            } else if (config.iframe) {
-              sendContextToClient(this, config, {});
-
-              const loadingIndicatorAutoHideEnabled =
-                GenericHelpers.getConfigValueFromObject(
-                  this.get(),
-                  'currentNode.loadingIndicator.hideAutomatically'
-                ) !== false;
-              if (loadingIndicatorAutoHideEnabled) {
-                this.set({ showLoadingIndicator: false });
-              }
-            }
+          const allIframes = Iframe.getAllIframes(node);
+          const getTargetIframe = (iframe) => {
+            return (e.source === iframe.contentWindow || (this.get().modalIframe && e.source === this.get().modalIframe.contentWindow))
           }
+          const iframe = allIframes.find(getTargetIframe);
+          if (iframe && iframe.luigiViewUrl) {
 
-          if ('luigi.show-loading-indicator' === e.data.msg) {
-            this.set({ showLoadingIndicator: true });
-          }
+            if ('luigi.navigate.ok' === e.data.msg) {
+              if (iframe.luigiNextViewUrl) {
```

Comment:
Here we should check if the e.origin matches the luigiNextViewUrl (the domain part).